content/developer/reference/backend/testing.rst

```diff
@@ -819,35 +840,39 @@ driven explicitly by the user) it's more complicated when running
 "test" tours, or when running tours through the test suite. In that
 case there are two main tricks:
 
-- Have a step with a ``run() { debugger; }`` action.
+- Have a step property ``break: true,`` in debug mode (debug=True).
+
+  This adds a debugger at the start of the step and then allows you
+  to put breakpoints in the desired locations.
+
+**Advantages**
+  - very simple
+  - the tour restarts as soon as you resume execution
+**Drawbacks**
+  - page interaction is limited as all javascript is blocked
+  - Have a step property ``pause: true,`` in debug mode (debug=True).
+
+  The tour will stop at the end of the step. This allows inspecting
+  and interacting with the page until the developer is ready to
+  resume by typing **play();** in console.
+
+**Advantages**
+  - allows interacting with the page
+  - no useless (for this situation) debugger UI
+  - Have a step with a ``run() { debugger; }`` action.
 
   This can be added to an existing step, or can be a new dedicated
   step. Once the step's **trigger** is matched, the execution will
   stop all javascript execution.
 
-  **Advantages**
-    - very simple
-    - the tour restarts as soon as you resume execution
-  **Drawbacks**
-    - page interaction is limited as all javascript is blocked
-    - debugging the inside of the tour manager is not very useful
-- Add a step with a trigger which never succeeds and a very long
-  ``timeout``.
-
-  The browser will wait for the **trigger** until the ``timeout``
-  before it fails the tour, this allows inspecting and interacting
-  with the page until the developer is ready to resume, by manually
-  enabling the **trigger** (a nonsense class is useful there, as it
-  can be triggered by adding the class to any visible element of the
-  page).
-
-  **Advantages**
-    - allows interacting with the page
-    - easy to apply to a step which times out (just add a long
-      ``timeout`` then look around)
-    - no useless (for this situation) debugger UI
-  **Drawbacks**
-    - more manual, especially when resuming
+**Advantages**
+  - simple
+  - the tour restarts as soon as you resume execution
+**Drawbacks**
+  - page interaction is limited as all javascript is blocked
+  - the debugger is triggered after trying to find the targeted
+elements.
+  - debugging the inside of the tour manager is not very useful
```

Comment:
I don't understand what you mean so I guess we should find a better sentence.
Do you mean that because of the debugger, the source page is opened with a breakpoint inside the Tour manager?

Reply:
My sentence was absolutely not necessary ...